### PR TITLE
refactor: updateURL function for improved maintainability (@anthonypz)

### PR DESCRIPTION
**Pull request by anthonypz**
### Description

This PR refactors the updateURL function to improve maintainability and readability.

Changes:
- Replaced if-statements with a settingsMap and forEach loop. This way, adding or removing settings is as simple as modifying the settingsMap array.
- Introduced lazy evaluation for share setting values using arrow functions on the `settingsMap` array. Now, values are only computed when they are actually needed, not when the settingsMap is created. Also, values that depend on the current state of the app such as the `getMode2(Config, currentQuote)` function can always get the most up-to-date value.
-  Created a separate updateShareModal function to handle UI updates for better separation of concerns.

I also tested the changes locally to make sure it worked just as before.

### Checks

- [x] Adding quotes?
  - [x] Make sure to include translations for the quotes in the description (or another comment) so we can verify their content.
- [x] Adding a language or a theme?
  - [x] If is a language, did you edit `_list.json`, `_groups.json` and add `languages.json`?
  - [x] If is a theme, did you add the theme.css?
    - Also please add a screenshot of the theme, it would be extra awesome if you do so!
- [x] Check if any open issues are related to this PR; if so, be sure to tag them below.
- [x] Make sure the PR title follows the Conventional Commits standard. (https://www.conventionalcommits.org for more info)
- [x] Make sure to include your GitHub username prefixed with @ inside parentheses at the end of the PR title.

Closes #

### CODE REVIEW

File: frontend/src/ts/modals/share-test-settings.ts [hunk 1]
```diff
@@ -27,58 +27,37 @@ type SharedTestSettings = [
...
+
+  const settingsMap = [
+    { key: "mode", value: () => Config.mode },
+    { key: "mode2", value: () => getMode2(Config, currentQuote) },
+    { key: "customText", value: () => CustomText.getData() },
+    { key: "punctuation", value: () => Config.punctuation },
+    { key: "numbers", value: () => Config.numbers },
+    { key: "language", value: () => Config.language },
+    { key: "difficulty", value: () => Config.difficulty },
+    { key: "funbox", value: () => Config.funbox },
```
Comment (Miodec) [inline frontend/src/ts/modals/share-test-settings.ts:42]: rename value to getValue

File: frontend/src/ts/modals/share-test-settings.ts [hunk 2]
```diff
@@ -27,58 +27,37 @@ type SharedTestSettings = [
...
-  }
-
-  if (getCheckboxValue("difficulty")) {
-    settings[6] = Config.difficulty;
-  }
-
-  if (getCheckboxValue("funbox")) {
-    settings[7] = Config.funbox;
-  }
+  settingsMap.forEach(({ key, value }, index) => {
```
Comment (Miodec) [inline frontend/src/ts/modals/share-test-settings.ts:45]: please use for of loop

**Comment by anthonypz** [comment]
Hey @Miodec, I've submitted the requested changes. Please let me know if there are any other changes you'd like to see.

**Review by Miodec — APPROVED** [review]
Thanks

**Pull request merged**